Marlin/src/inc/SanityCheck.h

```diff
@@ -2507,8 +2507,8 @@ static_assert(   _ARR_TEST(3,0) && _ARR_TEST(3,1) && _ARR_TEST(3,2)
     #error "Z_STEPPER_AUTO_ALIGN requires NUM_Z_STEPPER_DRIVERS greater than 1."
   #elif !HAS_BED_PROBE
     #error "Z_STEPPER_AUTO_ALIGN requires a Z-bed probe."
-  #elif ENABLED(Z_STEPPER_ALIGN_KNOWN_STEPPER_POSITIONS) && NUM_Z_STEPPER_DRIVERS != 3
-    #error "Z_STEPPER_ALIGN_KNOWN_STEPPER_POSITIONS requires NUM_Z_STEPPER_DRIVERS to be 3."
+  #elif ENABLED(Z_STEPPER_ALIGN_KNOWN_STEPPER_POSITIONS) && NUM_Z_STEPPER_DRIVERS < 3
+    #error "Z_STEPPER_ALIGN_KNOWN_STEPPER_POSITIONS requires NUM_Z_STEPPER_DRIVERS to be 3 or 4."
```

Comment:
Z_STEPPER_ALIGN_KNOWN_STEPPER_POSITIONS relies on calculating a plane and mapping screw heights onto it, to calculate adjustment distances. This relies on the bed being rigid, which is unlikely when using 4 steppers.

I think that known_stepper positions could be extended (with a little effort) to two steppers, but I don't know how you would make it account for the warping which necessarily occurs with 4 steppers.

Reply:
In theory I agree, but in practice it may not be as bad as we think it will be. I dont see a big enough problem to totally exclude it, but in time improvements based on the code done in RRF and Klipper explicitly for quad alignment may be looked at to improve things. There is a while separate class for quad z in klipper.

https://github.com/KevinOConnor/klipper/blob/master/klippy/extras/quad_gantry_level.py

But at a glance im not sure how much of it really diverges from what were already doing. Ill need to sit down and digest klipper a bit more at some point but time is short.

Regardless, thought process is that it should work enough to allow it to be used if the user really wants to knowing its not really intended for it.

Reply:
I really think the current Z_STEPPER_ALIGN_KNOWN_STEPPER_POSITIONS is completely invalid with four screws...

Based on Klipper code and commit comments, I think they are actually fitting curved lines out to the stepper locations, in order to achieve what a flat plane does with three steppers.

With the 3-stepper implementation, I think a perfect "potato chip" bed probed at 4 points would result in a best-fit plane that is perfectly flat and level, with no corrections necessary.

> Regardless, thought process is that it should work enough to allow it to be used if the user really wants to knowing its not really intended for it.

I tend to think this will only lead to disappointment, unless the individual is actually implementing a 4-point algorithm. In that case they will be very capable of removing a check.

Reply:
Seems like you would want to do something where you disable the stepper on one corner and "shimmy" the axis up and down rapidly by a small distance, like 0.2mm. This should have the effect of bringing the disabled stepper's screw into better accord with the enabled steppers. This same procedure could be applied to all 4 corners to alleviate warping caused by unwanted tension in the screws.